pyro/primitives.py

```diff
@@ -115,10 +114,8 @@ def sample(self, sample_shape=torch.Size()):
         if sample_shape:
             raise NotImplementedError
         subsample_size = self.subsample_size
-        if subsample_size is None or subsample_size > self.size:
-            subsample_size = self.size
-        if subsample_size == self.size:
-            result = torch.LongTensor(list(range(self.size)))
+        if subsample_size is None or subsample_size >= self.size:
+            result = torch.cumsum(torch.ones(self.size), dim=0, dtype=torch.long) - 1
```

Comment:
nit: Can we pull this out into something like
```py
# work around https://github.com/pytorch/pytorch/issues/11829
def _jit_compatible_arange(end, dtype=None):
    return torch.cumsum(torch.ones(end), dim=0, dtype=dtype) - 1
```

Reply:
We can also add this to `torch_patch` if we need it more widely!